Bundle pre-generated headers too

kornelski opened the pull request:
Pulling in llvm is quite slow, and very problematic in many environments, so I've made runtime bindgen behind a feature flag.

kornelski commented:
This may fix #109

iwillspeak commented:
Thanks. I like the idea but i'd prefer this to be opt out rather than opt in for `bindgen`. I think we should be using environment variables for this too rather than Cargo features. It seems more  of a built time problem rather than some configuration with extra features you enable. 

While I do like the upgrade to Rust 2018 could you move that part to a separate PR.

Thanks again for the contribution.

kornelski commented:
I've used Cargo feature to be able to avoid compiling bindgen, which requires clang. Env var can't disable a dependency.

Keats commented:
> The main reason to avoid bindgen is the heavy libclang/llvm dependency. It makes it harder to install, and takes long time to compile.

And is not really an option on Windows @ Azure Pipelines for example. I had to stick `syntect to a lower version to have CI working again. If it's not disabled by default, at least a feature that can be disabled like #113 would be great.

kornelski commented:
I've rebased this branch. I've also changed dependency's `features=[]` to `default-features=false`, because `features` is additive, so it didn't disable anything.

Cogitri commented:
Oh, my bad, sorry.

kornelski commented:
ping

Cogitri commented:
Although I'm not a direct user of this, onig is in my dependency chain (of Tau, https://gitlab.gnome.org/World/Tau) and this has worked fine for me, while avoiding me pulling in lots of unnecessary stuff.

iwillspeak commented:
Hopefully I should have some time over the next few days to work on `rust-onig` again. I'll have a go pulling together these PRs to refactor `bindgen` useage and publish a new crate version when I'm done.

iwillspeak commented:
Closing in favor of #126. Thanks!
